File: chapter-18-jpa-jooq/code-java/src/main/java/com/example/demo/entity/AuthorEntity.java
package com.example.demo.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import org.springframework.data.domain.Persistable;

@Entity
@Table(name = "author")
@Getter
@Setter
@NoArgsConstructor
public class AuthorEntity implements Persistable<Integer> {

    @Id
    private Integer id;

    @Column(name = "first_name")
    private String firstName;

    @Column(name = "last_name")
    private String lastName;

    public AuthorEntity(Integer id, String firstName, String lastName) {
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
    }

    @Override
    public boolean isNew() {
        return true;
    }
}

File: chapter-18-jpa-jooq/code-kotlin/src/main/kotlin/com/example/demo/entity/AuthorEntity.kt
package com.example.demo.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.springframework.data.domain.Persistable

@Entity
@Table(name = "author")
class AuthorEntity(
    @Id
    @get:JvmName("getIdentifier")
    var id: Int? = null,

    @Column(name = "first_name")
    var firstName: String? = null,

    @Column(name = "last_name")
    var lastName: String? = null
) : Persistable<Int> {

    override fun getId(): Int? = id

    override fun isNew(): Boolean = true
}
